python/lsst/multiprofit/fit_bootstrap_model.py

```diff
@@ -20,129 +20,156 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 from functools import cached_property
-from typing import Any, Mapping
+import logging
+from typing import Any, Mapping, Sequence
```

Comment:
For `Mapping`, yes, I have `source` as `astropy.table.Row | Mapping[str, Any]` since it ought to support other types of tables/catalogs (SourceCatalog, pandas, dict-like things, etc.), though admittedly I don't test to ensure that those all work and the last time I tried to see how many mypy errors I had it was such a long list that I put it off enabling it indefinitely.

Sequence, not so much but I find it annoying to have tuples excluded for container args that won't be mutated.